api-v2/src/plan/plan-not-found.error.ts

```diff
@@ -0,0 +1,5 @@
+export class PlanNotFoundError extends Error {
+  constructor(planId: number) {
+    super(`A Plan with the id, ${planId}, is not found`);
+  }
```

Comment:
should we instead return a ForbiddenError? theoretically the OwnPlanGuard should protect against users accessing plans that aren't theirs, but just in case

Reply:
So I throw this error specifically when the plan id literally doesn't exist in the db. This is a generic error thrown by services, https agnostic.

As for whether we should throw Forbidden in controllers when a plan doesn't exist, I think there is merit in doing so.

The sort of specificity we get from throwing a NotFound when a plan doesn't exist and Forbidden when a user is not authorized may make things less secure(since now the client knows if the id doesn't exist or if they are forbidden from accessing it).

At the same time the specificity allows us to handle the case where somehow a user that used to own a plan but deleted it accesses that plan. In a case like that we might want to display "Oh no, that plan doesn't exist anymore" rather than "Forbidden"? 

I could make an argument for handling it either way at the controller level, thoughts?

Reply:
Ohh i see, thanks. Yea ignore this